tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1422,6 +1423,32 @@ public function testCommentInTable(): void
         $table = $this->schemaManager->listTableDetails('table_with_comment');
         self::assertSame('Foo with control characters \'\\', $table->getComment());
     }
+
+    protected function diffSaveModeHandleOrphanForeignKeysBeforeIndexDrop(): SchemaDiff
+    {
+        if (! $this->schemaManager->getDatabasePlatform()->supportsForeignKeyConstraints()) {
+            $this->markTestSkipped('This test is only supported on platforms that have foreign keys.');
+        }
+
+        $primaryTable = new Table('test_save_mode_orphan_fk_primary');
+        $primaryTable->addColumn('id', 'integer');
+        $primaryTable->setPrimaryKey(['id']);
+        $primaryTable->addColumn('self', 'integer');
+        $primaryTable->addForeignKeyConstraint('test_save_mode_orphan_fk_primary', ['self'], ['id']);
+
+        $foreignTable = new Table('test_save_mode_orphan_fk_foreign');
+        $foreignTable->addColumn('fk', 'integer');
+        $foreignTable->addUniqueIndex(['fk']);
+        $foreignTable->addForeignKeyConstraint('test_save_mode_orphan_fk_primary', ['fk'], ['id']);
+
+        $toSchemaForeignTable = new Table('test_save_mode_orphan_fk_foreign');
+        $toSchemaForeignTable->addColumn('fk', 'integer');
+
+        $fromSchema = new Schema([$primaryTable, $foreignTable]);
+        $toSchema   = new Schema([$toSchemaForeignTable]);
+
+        return Comparator::compareSchemas($fromSchema, $toSchema);
```

Comment:
This isn't a functional test, it doesn't interact with a real database. It should execute the SQL on a real database and confirm that the code works as expected by introspecting the resulting schema.